### PR TITLE
Add product feature 

### PR DESCRIPTION
**Pull request by Iamyuvraj143**
Add Product feature added
<kpd>product_index</kpd>

<kpd>product_detail</kpd>

<kpd>add_new_product</kpd>

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		@user = Current.user
+		@product = Product.new
+	end
+
+	def show
+    @is_admin = is_admin?
+		@user = Current.user
+		@product = Product.find(params[:id])
+	end
+
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb]: Fix indentation and use `find_by` instead of `find`

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    def admin_authorization
+      if Current.user&.email != "admin@gmail.com"
+        redirect_to root_path
+        flash[:notice] = "You can not perform certain actions"
+      end
+    end
+
+    def is_admin?
+     Current.user&.email == "admin@gmail.com" ? true : false
+    end
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb]:
```suggestion
    def is_admin?
     Current.user&.email == "admin@gmail.com"
    end
```

Comment (kalpana19sharma) [reply app/controllers/products_controller.rb]: You can move this method to application controller if you need it in multiple controllers

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    if @product.update(product_params)
+      redirect_to @product
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user = Current.user
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb]: Why do you need `@user`?

File: app/models/product.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Product < ApplicationRecord
+	validates :name, presence: true
+	validates :price, presence: true, format: { with: /\A(\$)?(\d+)(\.|,)?\d{0,2}?\z/ } 
+	validates :quantity, presence: true, format: { with: /\A\d+\z/, message: "Integer only. No sign allowed." }, length: { maximum: 3}
+	validates :image, presence: true
+	validates :category, presence: true
+  validates :description, presence: true, length: { minimum: 100 }
```
Comment (kalpana19sharma) [inline app/models/product.rb]: Fix indentation

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: app/controllers/products_controller.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  def show
+    @is_admin = is_admin?
+  end
+
+  def create
+    @product = Product.new(product_params)
+
+    if @product.save
+      redirect_to @product
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb:15]: Set flash message

File: app/controllers/products_controller.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb]: Use single line syntax

File: db/migrate/20220414102910_create_products.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class CreateProducts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :products do |t|
```
Comment (kalpana19sharma) [inline db/migrate/20220414102910_create_products.rb:3]: Also push the schema.rb file changes

File: app/controllers/products_controller.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    if @product.update(product_params)
+      redirect_to @product
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @product.destroy
+    redirect_to root_path, status: :see_other
```
Comment (kalpana19sharma) [inline app/controllers/products_controller.rb:34]: What does status `:see_other` mean?

File: app/controllers/application_controller.rb [hunk 9]
```diff
@@ -7,4 +7,8 @@ def set_current_user
     Current.user = User.find_by(id: session[:user_id]) if session[:user_id]
   end
 
+  def is_admin?
```
Comment (kalpana19sharma) [inline app/controllers/application_controller.rb]: Do you need this method in controller as well? If not you can define this as a helper method and then directly call the method in views.

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: app/controllers/products_controller.rb (hunk 8 above)
Comment (Iamyuvraj143) [reply app/controllers/products_controller.rb:34]: The 303 See Other code is typically provided in response to a POST, PUT, or DELETE.
indicates that the redirects does not link to the requested resource itself, but to another page

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**